File: LeetCode/942.py
#Link: https://leetcode.com/problems/di-string-match/

class Solution:
    def diStringMatch(self, s: str) -> List[int]:
        minValue = 0
        maxValue = len(s)
        result = []
        for x in s:
            if x == 'I':
                result.append(minValue)
                minValue += 1
            else:
                result.append(maxValue)
                maxValue -= 1
        result.append(minValue)
        return result
        

File: LeetCode/cpp/942.cpp
//Link: https://leetcode.com/problems/di-string-match/

class Solution {
public:
    vector<int> diStringMatch(string s) {
        int min = 0 , max = s.length();
        vector<int> result;
        for(auto x : s) {
            if(x == 'I') result.push_back(min++);
            else result.push_back(max--);
        }
        result.push_back(min);
        return result;
    }
};
